sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -667,9 +667,15 @@ object JdbcUtils extends Logging {
     val getConnection: () => Connection = createConnectionFactory(options)
     val batchSize = options.batchSize
     val isolationLevel = options.isolationLevel
-    df.foreachPartition(iterator => savePartition(
-      getConnection, table, iterator, rddSchema, nullTypes, batchSize, dialect, isolationLevel)
-    )
+    if (options.numPartitions != null && options.numPartitions.toInt != df.rdd.getNumPartitions) {
+      df.repartition(options.numPartitions.toInt).foreachPartition(iterator => savePartition(
```

Comment:
Thank you for review, @srowen .

First, the property `numPartitions` already exists in [JDBCOptions.scala: Optional parameters only for reading](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala#L73-L83).
This PR makes that option meaningful during write operation.

Second, for dataframe usecases, we can call `repartition` before saving to manage this. Actually, I asked @lichenglin that way. But, the main purpose of issue requested by @lichenglin is about providing pure SQL way to control the number of partitions for writing. In SQL, this looks reasonable to me.